### DISCUSSION TITLE
How to `map send_text` to send `\e[5n`

### DISCUSSION THREAD

**Discussion by salmankhilji** [kovidgoyal/kitty#5097] [Q&A] [answered]
I am trying to get the `send_text` action to send the same sequence as the command `printf '\e[5n'`.  For some reason its not working.

1. Launch kitty like this:
  ```
  kitty -c NONE -o 'map ctrl+alt+a send_text normal \x1b[5n'
  ```

2. Within the new kitty window, bind shell command:
  ```
  bind -x '"\e[0n": echo FOO'
  ```

3. Verify that manually sending the command prints `FOO`:
  ```
  $ printf '\e[5n'
  FOO
  ```

4. Invoke the same using `ctrl+alt+a`.  Nothing happens.

As to what I am really trying to achieve is to get the shell to redraw the current line using the same hack as employed in the `fzf` shell integration, where the actual command is:

```
# To redraw line after fzf closes (printf '\e[5n')
bind '"\e[0n": redraw-current-line'
```

**Comment by kovidgoyal** [comment, marked as answer]
send_text works fine, as demonstrated by running:

kitty -o 'map f1 send_text all \x1b[5n' cat

and pressing f1

You will see

^[[5n 

as expected.
